src/iconv.jl

```diff
@@ -1,9 +1,61 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
 module iconv
-import Base: close, eof, flush, read, readall, write
+import Base: close, eof, flush, read, readall, write, show
 import Base.Libc: errno, strerror, E2BIG, EINVAL, EILSEQ
 export StringEncoder, StringDecoder, encode, decode
+export StringEncodingError, OutputBufferError, IConvError
+export InvalidEncodingError, InvalidSequenceError, IncompleteSequenceError
+
+
+abstract StringEncodingError
+
+# specified encodings are not supported
+type InvalidEncodingError <: StringEncodingError
+    msg::UTF8String
```

Comment:
As I've been changing it to use `message`, I had a thought, that maybe this loses some important flexibility.   If you use constructors, it you can have different messages, based on the number or types of the arguments to the constructors, but with this scheme, how would you manage that?
For this particular package, at least now, it probably doesn't matter, but I still am thinking about what we'd want for a much more flexible and efficient general error scheme for Julia.